itemadapter/utils.py

```diff
@@ -31,74 +37,93 @@ def _get_pydantic_model_metadata(item_model: Any, field_name: str) -> MappingPro
     field = item_model.model_fields[field_name]
 
     for attribute in [
-        "default",
-        "default_factory",
-        "alias",
         "alias_priority",
-        "validation_alias",
-        "serialization_alias",
-        "title",
-        "field_title_generator",
+        "alias",
+        "allow_inf_nan",
+        "annotation",
+        "coerce_numbers_to_str",
+        "decimal_places",
+        "default_factory",
+        "deprecated",
         "description",
+        "discriminator",
         "examples",
         "exclude",
-        "discriminator",
-        "deprecated",
-        "json_schema_extra",
+        "fail_fast",
+        "field_title_generator",
         "frozen",
-        "validate_default",
-        "repr",
-        "init",
+        "ge",
+        "gt",
         "init_var",
+        "init",
+        "json_schema_extra",
         "kw_only",
-        "pattern",
-        "strict",
-        "coerce_numbers_to_str",
-        "gt",
-        "ge",
-        "lt",
         "le",
-        "multiple_of",
-        "allow_inf_nan",
+        "lt",
         "max_digits",
-        "decimal_places",
-        "min_length",
         "max_length",
+        "min_length",
+        "multiple_of",
+        "pattern",
+        "repr",
+        "serialization_alias",
+        "strict",
+        "title",
         "union_mode",
-        "fail_fast",
+        "validate_default",
+        "validation_alias",
     ]:
         if hasattr(field, attribute) and (value := getattr(field, attribute)) is not None:
             metadata[attribute] = value
 
+    for attribute, default_value in [
+        ("default", PydanticUndefined),
+        ("metadata", []),
+    ]:
+        if hasattr(field, attribute) and (value := getattr(field, attribute)) != default_value:
+            metadata[attribute] = value
+
     return MappingProxyType(metadata)
 
 
 def _get_pydantic_v1_model_metadata(item_model: Any, field_name: str) -> MappingProxyType:
     metadata = {}
-    field = item_model.__fields__[field_name].field_info
+    field = item_model.__fields__[field_name]
+    field_info = field.field_info
+
+    for attribute in [
+        "default_factory",
+    ]:
+        value = getattr(field, attribute)
+        if value is not None:
+            metadata[attribute] = value
```

Comment:
This is probably done for consistency, but I think it still can be simplified, something like

```py
if value:= getattr(field, "default_factory") is not None:
    metadata["default_factory"] = value
```

You can also move it closer to defaults handling (closer to the end of the function). All super minor :)